feat(slack): add more information about critical findings

abant07 opened the pull request:
### Context

Currently, the slack report aggregates the checks that are found for a specific user by displaying the amount of passed checks, the amount of failed checks, and the number of resources that were checked. However, this is not very descriptive for the customer and it would be nice to have an additional section that lists the checks that are included in the findings (passed and failed)

If fixes an issue please add it with `Fix #4843 

### Description

I changed the slack.py file and the outputs.py file. In the outputs.py file I added an additional 2 key value pair in the stats hashmap (critical_failed_findings and critical_failed_findings). Since we cannot list all the checks that are included in a customer's account across all 3 providers, because some accounts will have thousands of checks. However, instead, I made it so that it lists the checks are marked as critical for both passed and failed findings. 

In the slack.py file I added two additional sections below the parameter args markdown, where it will iterate through all the finding check_names and display the critical passed and critical failed findings.
### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

abant07 committed:
Slack integration for critical medium high low mute alerts #4843

abant07 committed:
linting

abant07 committed:
linting

abant07 committed:
fixing severity

abant07 committed:
formatting

sergargar commented on prowler/lib/outputs/outputs.py:
```diff
@@ -82,9 +82,14 @@ def extract_findings_statistics(findings: list) -> dict:
         "total_pass": 0,
+        "total_muted_fail": 0,
         "total_fail": 0,
+        "total_muted_fail": 0,
```
```suggestion
```

sergargar commented:
Hi @abant07, thanks for your contribution!

However, for large reports we may encounter the following error:
`[ERROR] must be less than 3001 characters`

This is due to multiple critical passed/failed findings that are making the Slack message too long, we can maybe put the number of critical findings, what do you think?

abant07 commented:
Sure, I can put number of critical, high, medium, and low findings @sergargar  for pass and failed

abant07 committed:
critical count instead of listing findings

abant07 commented:
Hey @sergargar . You are right. I think even if I did add all the counts for high, critical, informational, low, and medium, it would go over the 3000 character limit. So I have just follow you and put the counts for critical for both pass and fail

abant07 committed:
linting

abant07 committed:
new counts for severities

abant07 committed:
lint

abant07 committed:
quotations

abant07 committed:
codecov

abant07 committed:
linting

I dont know why the accessanlyzer check is failing, I did not change anything related to that

abant07 commented:
Hey @sergargar do you know why the accessanalyzer check is failing, I did not edit it.

sergargar commented:
> Hey @sergargar do you know why the accessanalyzer check is failing, I did not edit it.

no worries, it is some issue affecting all PRs.

abant07 committed:
fixes

abant07 commented:
done. I realized there are no checks that have information severity. Unless I missed one. So for now, I put critical, high, medium, low

sergargar commented:
Thanks @abant07 ! Can you put the severities horizontally so the message has less height?

abant07 committed:
horizontal

MrCloudSec reviewed (approved):
Thanks for the contribution!
